app/main.py

```diff
@@ -1,25 +1,71 @@
 from abc import ABC
+from typing import Any
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: Any, name: str) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: Any, owner: Any) -> int:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: Any, value: int) -> None:
+        if value not in range(self.min_amount, self.max_amount + 1):
+            return
+        setattr(instance, self.protected_name, value)
 
 
 class Visitor:
-   pass
+    def __init__(self, name: str, age: int, weight: int, height: int) -> None:
+        self.name = name
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class SlideLimitationValidator(ABC):
-    pass
+    def __init__(self, age: int, weight: int, height: int) -> None:
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class ChildrenSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    age = IntegerRange(4, 14)
+    weight = IntegerRange(20, 50)
+    height = IntegerRange(80, 120)
 
 
 class AdultSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    age = IntegerRange(14, 60)
+    weight = IntegerRange(50, 120)
+    height = IntegerRange(120, 220)
 
 
 class Slide:
-    pass
+    def __init__(
+            self,
+            name: str,
+            limitation_class: SlideLimitationValidator
+    ) -> None:
+        self.name = name
+        self.limitation_class = limitation_class
+
+    def can_access(self, visitor: Visitor) -> bool:
+        if isinstance(self.limitation_class, AdultSlideLimitationValidator):
+            self.limitation_class = AdultSlideLimitationValidator
+        elif isinstance(self.limitation_class,
+                        ChildrenSlideLimitationValidator):
+            self.limitation_class = ChildrenSlideLimitationValidator
```

Comment:
why do u need this code? limitation_class is allready passed in this attribute

Reply:
We receive class instance to the init of Slide. When I initialize self.limitation_class with parameters I like initialize new instance from instance? or how to understand it correctly. Also when I write self.limitation_class(visitor.age, visitor.weight, visitor.height) Pycharm underline me that it's not correct